mesonbuild/dependencies/dev.py

```diff
@@ -568,6 +568,18 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
 
         m = self.env.machines[self.for_machine]
 
+        if m.is_android():
+            if 'version' in kwargs:
+                mlog.error('Can\'t set version filter for jni on Android')
+                self.is_found = False
+                return
+            if 'modules' in kwargs:
+                mlog.error('Can\'t depend on jni modules on Android')
```

Comment:
maybe: "Android does not provide JNI modules"

I'm also not sure that this is a good place for an `error`. I'm trying to decide if I think `warning` or `debug` is better

Reply:
Moved this to a debug message, this way you can probably do something like

```meson
cfg = configuration()
jni_dep = dependency('jni', modules: ['jvm'], required: false)
cfg.set('JNI_HAS_JVM_MODULE', jni_dep.found())
if not jni_dep.found()
  jni_dep = dependency('jni')
endif
```

if you can provide a fallback path